crml/nft/src/benchmarking.rs

```diff
@@ -0,0 +1,274 @@
+
+	// Schema with max. attributes
+	let schema = (0..MAX_SCHEMA_FIELDS)
+		.map(|i| {
+			let mut v = b"probably-a-decent-upper-length-for-an-attribte-name".to_vec();
+			v.push(i as u8);
+			(v, NFTAttributeValue::String(Default::default()).type_id())
+		})
+		.collect::<NFTSchema>();
+
+	// Royalties with max. entitled addresses
+	let royalties = RoyaltiesSchedule::<T::AccountId> {
+		entitlements: (0..MAX_ENTITLEMENTS)
+			.map(|_| (creator.clone(), Permill::from_percent(1)))
+			.collect::<Vec<(T::AccountId, Permill)>>(),
+	};
+
+	return (collection_id, schema, royalties);
+}
+
+// Create a token for benchmarking
+fn setup_token<T: Trait>(owner: T::AccountId) -> CollectionId {
+	let creator: T::AccountId = whitelisted_caller();
+	let (collection_id, schema, royalties) = setup_collection::<T>(creator.clone());
+	let _ = <Nft<T>>::create_collection(
+		RawOrigin::Signed(creator.clone()).into(),
+		collection_id.clone(),
+		schema.clone(),
+		Some(royalties.clone()),
+	)
+	.expect("created collection");
+	// all attributes max. length
+	let attributes = schema
+		.iter()
+		.map(|_| NFTAttributeValue::String([1_u8; 140_usize].to_vec()))
+		.collect::<Vec<NFTAttributeValue>>();
+	let _ = <Nft<T>>::create_token(
+		RawOrigin::Signed(creator.clone()).into(),
+		collection_id.clone(),
+		owner.clone(),
+		attributes.clone(),
+		Some(royalties),
+	)
+	.expect("created token");
+
+	return collection_id;
+}
+
+benchmarks! {
+	_{}
+
+	create_collection {
+		let creator: T::AccountId = account("creator", 0, 0);
+		let (collection_id, schema, royalties) = setup_collection::<T>(creator.clone());
+
+	}: _(RawOrigin::Signed(creator.clone()), collection_id.clone(), schema, Some(royalties))
+	verify {
+		assert_eq!(<Nft<T>>::collection_owner(&collection_id), Some(creator));
+	}
+
+	create_token {
+		let creator: T::AccountId = whitelisted_caller();
+		let owner: T::AccountId = account("owner", 0, 0);
+
+		let (collection_id, schema, royalties) = setup_collection::<T>(creator.clone());
+		let _ = <Nft<T>>::create_collection(RawOrigin::Signed(creator.clone()).into(), collection_id.clone(), schema.clone(), Some(royalties.clone())).expect("created collection");
+		// all attributes max. length
+		let attributes = schema.iter().map(|_| NFTAttributeValue::String([1_u8; 140_usize].to_vec())).collect::<Vec<NFTAttributeValue>>();
+
+	}: _(RawOrigin::Signed(creator.clone()), collection_id.clone(), owner.clone(), attributes, Some(royalties))
+	verify {
+		assert_eq!(<Nft<T>>::token_owner(&collection_id, T::TokenId::from(0_u32)), owner);
+	}
+
+	transfer {
+		let owner: T::AccountId = account("owner", 0, 0);
+		let collection_id = setup_token::<T>(owner.clone());
+		let new_owner: T::AccountId = account("new_owner", 0, 0);
+		let token_id = T::TokenId::from(0_u32);
+
+	}: _(RawOrigin::Signed(owner.clone()), collection_id.clone(), token_id, new_owner.clone())
+	verify {
+		assert_eq!(<Nft<T>>::token_owner(&collection_id, token_id), new_owner);
+	}
+
+	burn {
```

Comment:
Looks like the expense of burn would be variable based on the number of tokens of a particular id that an owner has collected:
`			<CollectedTokens<T>>::mutate(&collection_id, current_owner, |tokens| {
				tokens.retain(|t| t != &token_id)
			});
`
Can you make the benchmark take that count into account or test a worst case/average scenario?